jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -878,15 +878,16 @@ def _update_stretch_histogram(self, msg={}):
             if not len(self.layer.selected_obj[0]):
                 return
             # multiselect case (but we won't check multiselect since the selection can lag behind)
-            data = self.layer.selected_obj[0][0].layer
+            layer = self.layer.selected_obj[0][0]
         else:
-            data = self.layer.selected_obj[0].layer
+            layer = self.layer.selected_obj[0]
+        data = layer.layer
 
         if isinstance(data, GroupedSubset):
             # don't update histogram for subsets:
             return
 
-        comp = data.get_component(data.main_components[0])
+        comp = data.get_component(layer.state.attribute)

```

Comment:
Is this really equivalent to `main_component[0]`?

Reply:
No, but I think its what we actually want.  Shouldn't this always be the _plotted_ data?

Reply:
I dunno. Further down, there is a block that calls `data.get_component` under some condition. Will that still be consistent?

Reply:
That logic probably also could use rewriting to be more specific as well (currently only applies to 2d spectral viewers, afaik), but should not affect this at all (that is for the x-y limit determination, this is for the actual "color" axis of the image which is what should be plotted on the histogram).  Feel free to find a case that breaks if you have any concerns!

Reply:
`layer.state.attribute` is the requested column to be plotted, `data.main_components[0]` is just the first "main" component (which probably is the _default_ column to be plotted, but is not general enough).  For the histogram, we definitely want to plot the same data in the histogram as is shown in the image viewer itself, so I think the previous assumption is not general enough.